diff --git a/examples/napi/__tests__/values.spec.ts b/examples/napi/__tests__/values.spec.ts
index 8e0fa505..ef475ab4 100644
--- a/examples/napi/__tests__/values.spec.ts
+++ b/examples/napi/__tests__/values.spec.ts
@@ -225,9 +225,9 @@ import {
   callThenOnPromise,
   callCatchOnPromise,
   callFinallyOnPromise,
-  StructuredKind,
+  //StructuredKind,
   validateStructuredEnum,
-  StructuredKindLowercase,
+  //StructuredKindLowercase,
   validateStructuredEnumLowercase,
   createArraybuffer,
   getBufferSlice,
@@ -248,7 +248,7 @@ import {
   getMappingWithHasher,
   getIndexMappingWithHasher,
   passSetWithHasherToJs,
-  Rule,
+  //Rule,
   callRuleHandler,
   acceptStream,
   createReadableStream,
@@ -256,7 +256,7 @@ import {
   spawnThreadInThread,
   esmResolve,
   mergeTupleArray,
-  TupleToArray,
+  //TupleToArray,
   ClassInArray,
   getClassFromArray,
   extendsJavascriptError,
