diff --git a/examples/napi/index.cjs b/examples/napi/index.cjs
index aa786e27..435344a5 100644
--- a/examples/napi/index.cjs
+++ b/examples/napi/index.cjs
@@ -105,9 +105,22 @@ function requireNative() {
     }
   } else if (process.platform === 'win32') {
     if (process.arch === 'x64') {
-      if (process.config?.variables?.shlib_suffix === 'dll.a' || process.config?.variables?.node_target_type === 'shared_library') {
+      if (process.config?.variables?.shlib_suffix === 'dll.a' || process.config?.variables?.node_target_type === 'shared_library' || process.config?.variables?.shlib_suffix === 'so') {
         try {
-        return require('./example.win32-x64-gnu.node')
+          let abi = "";
+          switch (process.env.MSYSTEM) {
+            // TODO need to check ABI compatibility
+            case 'UCRT64':
+            case 'MINGW64':
+              abi = 'gnu'
+              break;
+            case 'CLANG64':
+              abi = 'gnullvm'
+              break;
+            default:
+              throw new Error('Unsupported platform')
+          }
+        return require(`./example.win32-x64-${abi}.node`)
       } catch (e) {
         loadErrors.push(e)
       }
@@ -156,7 +169,14 @@ function requireNative() {
       }
     } else if (process.arch === 'arm64') {
       try {
-        return require('./example.win32-arm64-msvc.node')
+        if (process.config?.variables?.shlib_suffix === 'dll.a' || process.config?.variables?.node_target_type === 'shared_library' || process.config?.variables?.shlib_suffix === 'so') {
+          if (process.env.MSYSTEM !== 'CLANGARM64') {
+            throw new Error('Unsupported platform')
+          }
+          return require('./example.win32-arm64-gnullvm.node')
+        } else {
+          return require('./example.win32-arm64-msvc.node')
+        }
       } catch (e) {
         loadErrors.push(e)
       }
