--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,4 +1,4 @@
-[target.'cfg(target_env = "gnu")']
+[target.'cfg(all(target_env = "gnu", target_os = "linux"))']
 rustflags = ["-C", "link-args=-Wl,-z,nodelete"]

 [target.x86_64-pc-windows-msvc]
@@ -10,7 +10,10 @@ rustflags = ["-C", "target-feature=+crt-static"]
 [target.aarch64-pc-windows-msvc]
 rustflags = ["-C", "target-feature=+crt-static"]

-[target.'cfg(target_os = "windows")']
+[target.x86_64-pc-windows-gnu]
+rustflags = [ "-C", "link-arg=-ladvapi32" ]
+
+[target.'cfg(all(target_os = "windows", target_env = "msvc"))']
 rustflags = [
   # Disables linking against the default Universal C Runtime (libucrt.lib). This prevents conflicts if another CRT library is linked manually.
   "-C",
